core/bin/external_node/src/config/mod.rs

```diff
@@ -1341,6 +1350,24 @@ impl ExternalNodeConfig {
             tree_component: TreeComponentConfig { api_port: None },
         }
     }
+
+    /// Returns a verified diamond proxy address.
+    /// If local configuration contains the address, it will be checked against the one returned by the main node.
+    /// Otherwise, the remote value will be used. However, using remote value has trust implications for the main
+    /// node so relying on it solely is not recommended.
+    pub fn diamond_proxy_address(&self) -> anyhow::Result<Address> {
+        let remote_diamond_proxy_addr = self.remote.diamond_proxy_addr;
+        let diamond_proxy_addr = if let Some(addr) = self.optional.contracts_diamond_proxy_addr {
+            addr
+        } else {
+            tracing::info!(
```

Comment:
Nit: I would move this log to the `anyhow::ensure!` check above so that it's too not spammed on multiple calls. Then, the method can be simplified to

```rust
self.optional
    .contracts_diamond_proxy_addr
    .unwrap_or(self.remote.diamond_proxy_addr)
```